cpp/tests/community/weighted_matching_test.cpp

```diff
@@ -0,0 +1,175 @@
+#include "utilities/test_graphs.hpp"
+
+#include <cugraph/algorithms.hpp>
+#include <cugraph/edge_partition_view.hpp>
+#include <cugraph/edge_property.hpp>
+#include <cugraph/edge_src_dst_property.hpp>
+#include <cugraph/graph_functions.hpp>
+#include <cugraph/graph_view.hpp>
+#include <cugraph/utilities/high_res_timer.hpp>
+
+#include <raft/random/rng_state.hpp>
+
+#include <gtest/gtest.h>
+
+#include <chrono>
+#include <iostream>
+#include <random>
+
+struct WeightedMatching_UseCase {
+  bool check_correctness{true};
+};
+
+template <typename input_usecase_t>
+class Tests_SGWeightedMatching
+  : public ::testing::TestWithParam<std::tuple<WeightedMatching_UseCase, input_usecase_t>> {
+ public:
+  Tests_SGWeightedMatching() {}
+
+  static void SetUpTestCase() {}
+  static void TearDownTestCase() {}
+
+  virtual void SetUp() {}
+  virtual void TearDown() {}
+
+  template <typename vertex_t, typename edge_t, typename weight_t, typename result_t>
+  void run_current_test(std::tuple<WeightedMatching_UseCase, input_usecase_t> const& param)
+  {
+    auto [weighted_matching_usecase, input_usecase] = param;
+
+    raft::handle_t handle{};
+    HighResTimer hr_timer{};
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());
+      hr_timer.start("Construct graph");
+    }
+
+    constexpr bool multi_gpu = false;
+
+    bool test_weighted    = true;
+    bool renumber         = true;
+    bool drop_self_loops  = false;
+    bool drop_multi_edges = false;
+
+    auto [sg_graph, sg_edge_weights, sg_renumber_map] =
+      cugraph::test::construct_graph<vertex_t, edge_t, weight_t, false, multi_gpu>(
+        handle, input_usecase, test_weighted, renumber, drop_self_loops, drop_multi_edges);
+
+    std::tie(sg_graph, sg_edge_weights, sg_renumber_map) = cugraph::symmetrize_graph(
+      handle, std::move(sg_graph), std::move(sg_edge_weights), std::move(sg_renumber_map), false);
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());
+      hr_timer.stop();
+      hr_timer.display_and_clear(std::cout);
+    }
+
+    auto sg_graph_view = sg_graph.view();
+    auto sg_edge_weight_view =
+      sg_edge_weights ? std::make_optional((*sg_edge_weights).view()) : std::nullopt;
+
+    rmm::device_uvector<vertex_t> d_partners(0, handle.get_stream());
+    weight_t total_matching_weights;
+
+    std::forward_as_tuple(d_partners, total_matching_weights) =
+      cugraph::approximate_weighted_matching<vertex_t, edge_t, weight_t, multi_gpu>(
+        handle, sg_graph_view, (*sg_edge_weights).view());
+
+    if (weighted_matching_usecase.check_correctness) {
+      auto h_partners                = cugraph::test::to_host(handle, d_partners);
+      auto constexpr invalid_partner = cugraph::invalid_vertex_id<vertex_t>::value;
+
+      std::for_each(h_partners.begin(), h_partners.end(), [&invalid_partner, h_partners](auto& v) {
+        if (v != invalid_partner) ASSERT_TRUE(h_partners[h_partners[v]] == v);
+      });
```

Comment:
Can't we at least check that this satisfies requirements to be a proper matching?

Reply:
Do you mean [perfect matching](https://en.wikipedia.org/wiki/Perfect_matching)?
It's not guarantee to a perfect matching.

Reply:
"In graph theory, a matching in a graph is a set of edges that do not have a set of common vertices." Shouldn't this be enforced even for approximate matching?

Reply:
This enforced is already enforced by the check  `h_partners[h_partners[v]] == v`